integration/spark/src/main/scala/org/apache/spark/sql/catalyst/CarbonParserUtil.scala

```diff
@@ -213,8 +207,8 @@ object CarbonParserUtil {
 
     // Process spatial index property
     val indexFields = processSpatialIndexProperty(tableProperties, fields)
-    val allFields = fields ++ indexFields
 
+    val allFields = (fields ++ indexFields).distinct
```

Comment:
why distinct ? have already checked that indexFields are not in actual fields. right ? If required please add the comments here.